### PR TITLE
Refactorings & improvments for AssemblyScript contracts

### PR DESCRIPTION
**Pull request by MaxGraey**
Similar changes and improvements mostly exposed in PR for `as-contract`: https://github.com/Stefie/as-contract/pull/1

### CODE REVIEW

**Comment by Stefie** [comment]
Thanks a lot! I'm just finishing sth. here, I'll have a look in a bit.

File: contracts/assemblyscript/incrementer/assembly/index.ts [hunk 1]
```diff
@@ -24,28 +23,31 @@ enum Action {
...
+  const value = new Uint8Array(0);
+  const counter = getStorage(COUNTER_KEY);
+  const dataCounter = new DataView(counter.buffer);
   const counterValue: u32 = dataCounter.byteLength ? dataCounter.getUint32(0, true) : 0;
 
   // Get action from first byte of the input U8A
   switch (input[0]) {
-    case Action.Inc:
-      // read 4 bytes (u32) from storageBuffer with offset 1 
-      const by: u32 = load<u32>(input.dataStart, 1);
```
Comment (MaxGraey) [inline contracts/assemblyscript/incrementer/assembly/index.ts:36]: Btw I'm wondering why offset equal to `1` byte? Is it correct?

Comment (MaxGraey) [reply contracts/assemblyscript/incrementer/assembly/index.ts:36]: ok. It seems I understand. First byte encodes type of Action

Comment (MaxGraey) [reply contracts/assemblyscript/incrementer/assembly/index.ts:36]: Will be great explain this in comment I guess

Comment (Stefie) [reply contracts/assemblyscript/incrementer/assembly/index.ts:36]: Yeah, that's also on the list: Add proper comments. As I said in the other PR, this is all still pretty rough on the edges and not meant to be presented as proper examples yet. There's still a lot of work to do.

File: contracts/assemblyscript/flipper/assembly/lib.ts [hunk 2]
```diff
@@ -1,50 +1,55 @@
+/// <reference path="../node_modules/assemblyscript/std/assembly/index.d.ts" />
```
Comment (Stefie) [inline contracts/assemblyscript/flipper/assembly/lib.ts:1]: Nice! And it seems like it's definitely worth reading through that file btw. :)

File: contracts/assemblyscript/flipper/assembly/index.ts [hunk 3]
```diff
@@ -21,46 +21,44 @@ enum Action {
...
   const flipperValue: u8 = dataFlipper.byteLength ? dataFlipper.getUint8(0) : 0;
 
   // Get action from first byte of the input U8A
   switch (input[0]) {
-    case Action.Flip:
-      const newFlipperBool: bool = !<bool>flipperValue;
-      const newFlipperU8a: Uint8Array = new Uint8Array(1);
-      newFlipperU8a[0] = newFlipperBool;
-      setStorage(FLIPPER_KEY, newFlipperU8a)
+    case Action.Flip: {
```
Comment (Stefie) [inline contracts/assemblyscript/flipper/assembly/index.ts]: Arrows in a switch case are rather non-standard and not necessary so I'd prefer to not add them here & in the other example.

**Review by Stefie — CHANGES_REQUESTED** [review]
Thanks again for moving the changes over here.
As for the linting changes: It's possibly going to get messed up again until we added a linter because as I said, it's still WIP, but thanks for caring anyways!

File: contracts/assemblyscript/flipper/assembly/index.ts (hunk 3 above)
Comment (MaxGraey) [reply contracts/assemblyscript/flipper/assembly/index.ts]: I removed brackets

**Review by Stefie — APPROVED** [review]

**Pull request merged**